packages/core-ethereum/src/index.ts

```diff
@@ -246,17 +246,37 @@ export default class HoprCoreEthereum extends EventEmitter {
     return this.redeemingAll
   }
 
-  private async redeemAllTicketsInternalLoop(): Promise<void> {
-    try {
-      for (const ce of await this.db.getChannelsTo(this.publicKey.toAddress())) {
-        await this.redeemTicketsInChannel(ce)
+  private redeemAllTicketsInternalLoop(): Promise<void> {
+    return new Promise<void>(async (resolve) => {
```

Comment:
Don't create such a complex return value inline with the return statement. Split them apart. The whole logic of the new approach needs to be documented better, the use of the state object is unclear.